templates/glossary.md

```diff
@@ -134,9 +137,27 @@ Examples of such commands are the `def`, `theorem`, `constant` or `example` comm
 
 Further detail can be found in the [Lean documentation](https://leanprover.github.io/lean4/doc/declarations.html#basic-declarations).
 
+### dependent type theory
+
+A [type theory](#type-theory) in which one additionally may have types which depend on a parameter, such as the type of "days of a calendar month", whose specific values depend on the particular month, given that different months have different numbers of days.
+Further examples may be found in the resources below.
+Lean's implementation of dependent type theory is based on what is known as the *Calculus of Constructions*, allowing for its use both in complex mathematical reasoning as well as software verification.
```

Comment:
Here I was going to add some additional notes but I think if we have entries for pi type and sigma type (which probably we should) that the right thing to do will be to then just mention those two here, but thoughts welcome.